src/mlpack/methods/reinforcement_learning/training_config.hpp

```diff
@@ -0,0 +1,124 @@
+/**
+ * @file training_config.hpp
+ * @author Shangtong Zhang
+ *
+ * This file is the implementation of TrainingConfig class,
+ * which contains hyper-parameters for training RL agent.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_RL_TRAINING_CONFIG_HPP
+#define MLPACK_METHODS_RL_TRAINING_CONFIG_HPP
+
+namespace mlpack {
+namespace rl {
+
+class TrainingConfig
```

Comment:
I like the approach, but it's not conform with what's proposed here: https://github.com/mlpack/mlpack/pull/1044#issuecomment-314816704. I think for now this is fine. Not sure adding each parameter to the constructor is helpful here. What be great to get @rcurtin's view on this.